common/utils/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -24,20 +24,14 @@ import java.util.ConcurrentModificationException
 
 import scala.jdk.CollectionConverters._
 
-class SparkException(
+class SparkException private(
```

Comment:
This constructor was exposed in 3.2, right ? Making this a backwardly incompatible change ?

Reply:
> This constructor was exposed in 3.2, right ?

Yep, it exists for a while.

> Making this a backwardly incompatible change ?

Yes, but `SparkException` is supposed to be raised by Spark mostly. And if we are going to migrate on new error framework, I do believe we should eliminate Spark's exception without error classes/conditions at the end.

> Users can raise SparkException for a variety of reasons - including in test code.

Users might use special error conditions for tests, for instance `_LEGACY_ERROR_USER_RAISED_EXCEPTION` or `USER_RAISED_EXCEPTION`.

Reply:
My point being, this is not a backward compatible change right ?
Any user code which was using this constructor will now fail to compile

Reply:
> My point being, this is not a backward compatible change right ?

Right.

> Any user code which was using this constructor will now fail to compile

Sure if user's code raises `SparkException` w/ arbitrary error message, it will fail, and we force to migrate on error classes/conditions - so, the user has to migrate its code.

Reply:
I will let the other reviewers comment on this more, since I am not following the evolution of error classes closely.
But I would encourage ensuring this is in line with [versioning policy](https://spark.apache.org/versioning-policy.html), and we minimize breaking api's.